core/org.wso2.carbon.bootstrap/src/test/java/org/wso2/carbon/bootstrap/BootstrapTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.bootstrap;
+
+import org.junit.rules.TemporaryFolder;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class BootstrapTest {
+    private Bootstrap bootstrap = new Bootstrap();
+    private final Set<URL> classpath = new LinkedHashSet<URL>();
+    private File libFile;
+
+    /**
+     * Test if loadClass throws ClassNotFoundException for non-existent class
+     *
+     * @throws Exception
```

Comment:
throws exception on which scenario ? the java doc is not complete. Same thing should be fixed on other methods too.

Reply:
Fixed in 1fbd44f